Modules/Sources/PointOfSale/Models/PointOfSaleAggregateModel.swift

```diff
@@ -620,6 +623,24 @@ private extension PointOfSaleAggregateModel {
     }
 }
 
+// MARK: - Catalog Sync State Observation
+
+private extension PointOfSaleAggregateModel {
+    private func publishCatalogSyncState() {
+        guard let coordinator = catalogSyncCoordinator else { return }
+
+        Task { @MainActor [weak self] in
+            guard let self else { return }
+
+            catalogSyncState = await coordinator.lastFullSyncState(for: siteID)
+
+            for await state in coordinator.fullSyncStateStream {
+                catalogSyncState = state
+            }
+        }
```

Comment:
I think this task might never end, and therefore be a memory leak. It doesn't capture self, so no retain cycle, but we'll have the task hanging out for each time we leave and enter POS. Perhaps we should keep hold of it, and cancel it when we deinit?

Reply:
Yes, it doesn't end since we want an endless stream. Good point on a memory leak. Need to be careful with this 🤔

Reply:
I think it would be fine to just keep hold of the task in a property, and cancel/finish when we deinit the model, wouldn't it?